internal/tools/jsonschema_patch.sed

```diff
@@ -0,0 +1,4 @@
+# go-jsonschema always generates patternProperties as
+# map[string]interface{}, for more specific types, they must
+# be replaced here
+s+type Headers.*+type Headers map[string]string+g
```

Comment:
nit: I think an actual patch to be applied by `patch` would be more resilient here (not a fan of sed specially after open-telemetry/opentelemetry-collector-contrib/issues/23981)